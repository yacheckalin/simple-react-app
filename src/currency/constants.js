export const DEFAULT_EXCHANGE_ONE_VALUE = ["USD", { disabled: false }];
export const DEFAULT_EXCHANGE_TWO_VALUE = ["USD", { disabled: false }];
export const DEFAULT_EXCHANGE_AMOUNT = 0;
export const DEFAULT_EXCHANGE_SUMMARY = null;
export const DEFAULT_CURRENCY_LIST_ONE = [];
export const DEFAULT_CURRENCY_LIST_TWO = [];
export const DEFAULT_CURRENCY_LIST = [];
export const DEFAULT_CURRENCY_LIST_DATA = [];

export const SHORT_DATE_FORMAT = "YYYY-MM-DD";
