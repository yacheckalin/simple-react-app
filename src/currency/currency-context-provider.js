import React, { useContext, useState } from "react";

export const CurrencyContext = React.createContext();
export const useCurrencyContext = () => useContext(CurrencyContext);

let currencyData = [
  {
    favorite: false,
    asset: "EUR/USD",
    quote: "1.12",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/JPY",
    quote: "107.95",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "GBP/USD",
    quote: "1.238",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "AUD/USD",
    quote: "0.7048",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/CHF",
    quote: "0.9888",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
  {
    favorite: false,
    asset: "USD/CAD",
    quote: "1.31",
    date: new Intl.DateTimeFormat("en-US", {
      year: "numeric",
      month: "long",
      day: "2-digit",
    }).format(Date.now()),
  },
];

const CurrencyContextProvider = ({ children }) => {
  const [data, setData] = useState(currencyData);

  const handleFavoriteChange = (index) => {
    if (currencyData[index]) {
      currencyData[index].favorite = !currencyData[index].favorite;
      setData([...currencyData]);
    }
  };
  return (
    <CurrencyContext.Provider
      value={{ data, setFavorite: handleFavoriteChange }}
    >
      {children}
    </CurrencyContext.Provider>
  );
};

export default CurrencyContextProvider;
