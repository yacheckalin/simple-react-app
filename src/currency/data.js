export const currencyData = [
  {
    favorite: false,
    asset: "EUR/USD",
    quote: "1.12",
    date: new Date("2019-04-08"),
  },
  {
    favorite: false,
    asset: "USD/JPY",
    quote: "107.95",
    date: new Date("2019-04-11"),
  },
  {
    favorite: false,
    asset: "GBP/USD",
    quote: "1.238",
    date: new Date("2019-04-18"),
  },
  {
    favorite: false,
    asset: "AUD/USD",
    quote: "0.7048",
    date: new Date("2020-04-28"),
  },
  {
    favorite: false,
    asset: "USD/CHF",
    quote: "0.9888",
    date: new Date("2020-04-08"),
  },
  {
    favorite: false,
    asset: "USD/CAD",
    quote: "1.31",
    date: new Date("2019-02-08"),
  },
];
