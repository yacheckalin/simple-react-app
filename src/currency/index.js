export { default } from "./currency-container";
