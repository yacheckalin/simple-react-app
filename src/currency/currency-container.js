import React from "react";
import styled from "styled-components";

import { useCurrencyContext } from "./currency-context-provider";

const Table = styled.table`
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  border-spacing: 0;
  background-color: white;
  max-width: 750px;
  overflow: hidden;
  border-top-right-radius: 10px;
  width: 100%;

  thead {
    background-color: #1a237e;
  }
  th,
  td {
    padding: 20px 0;
    text-align: left;
  }
  th {
    color: white;
  }
  td {
    border-bottom: 1px solid #e5e5e5;
    width: 200px;
  }
  td:first-child,
  td:last-child {
    border: 0;
  }
  td:first-child {
    width: 60px;
    color: #1a237e;
    text-align: right;
    padding-right: 10px;
  }
  td:last-child {
    width: 40px;
  }
  tr:first-child td {
    padding-top: 40px;
  }
  tr:last-child td {
    padding-bottom: 40px;
    border: 0;
  }
`;

const CurrencyContainer = () => {
  const { data, setFavorite } = useCurrencyContext();

  const EmptyStar = () => <>&#9734;</>;
  const FullfilledStar = () => <>&#9733;</>;

  return (
    <Table>
      <thead>
        <tr>
          <th>&nbsp;</th>
          <th>Валютная Пара</th>
          <th>Катировка</th>
          <th>Дата Получения</th>
          <th>&nbsp;</th>
        </tr>
      </thead>
      <tbody>
        {data.map(({ favorite, asset, quote, date }, index) => (
          <tr key={index} onClick={(e) => setFavorite(index)}>
            <td>{!favorite ? <EmptyStar /> : <FullfilledStar />}</td>
            <td>{asset}</td>
            <td>{quote}</td>
            <td>{date}</td>
            <td>&nbsp;</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default CurrencyContainer;
