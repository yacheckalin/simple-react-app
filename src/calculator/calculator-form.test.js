import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CalculatorForm from "./calculator-form";

import * as CurrencyContext from "../currency/currency-context-provider";
import Button from "./calculator-form-button";

describe("<CalculatorForm />", () => {
  const contextValues = {
    currencyListOne: ["USD", "EUR", "GMB", "CAD", "JPY"],
    currencyListTwo: ["USD", "EUR", "GMB", "CAD", "JPY"],
    exchangeOne: "USD",
    exchangeTwo: "USD",
    exchangeAmount: 0,
    setExchangeAmount: jest.fn(),
    setExchangeTwo: jest.fn(),
    validateExchangeOne: jest.fn(),
  };
  beforeEach(() => {
    jest
      .spyOn(CurrencyContext, "useCurrencyContext")
      .mockImplementation(() => contextValues);
  });
  test("renders and displays properly", () => {
    const wrapper = shallow(<CalculatorForm />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
