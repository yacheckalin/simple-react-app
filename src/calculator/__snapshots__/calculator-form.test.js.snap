// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorForm /> renders and displays properly 1`] = `
<styled.form>
  <input
    onChange={[Function]}
    type="text"
    value={0}
  />
  <select
    defaultValue="USD"
    onChange={[Function]}
  >
    <option
      key="0"
      value="USD"
    >
      USD
    </option>
    <option
      key="1"
      value="EUR"
    >
      EUR
    </option>
    <option
      key="2"
      value="GMB"
    >
      GMB
    </option>
    <option
      key="3"
      value="CAD"
    >
      CAD
    </option>
    <option
      key="4"
      value="JPY"
    >
      JPY
    </option>
  </select>
  <select
    defaultValue="USD"
    onChange={[Function]}
  >
    <option
      key="0"
      value="USD"
    >
      USD
    </option>
    <option
      key="1"
      value="EUR"
    >
      EUR
    </option>
    <option
      key="2"
      value="GMB"
    >
      GMB
    </option>
    <option
      key="3"
      value="CAD"
    >
      CAD
    </option>
    <option
      key="4"
      value="JPY"
    >
      JPY
    </option>
  </select>
  <Button
    disabled={true}
  >
    Расчитать
  </Button>
</styled.form>
`;
